HTTPS question

#### Description
Hello, I've got a weird comportment with https, just imagine this following yaml:
```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: f5-dev-it-transverse
  namespace: dev-it-transverse
  annotations:
    kubernetes.io/ingress.class: "f5"
    virtual-server.f5.com/ip: "*******"
    virtual-server.f5.com/partition: "******"
    virtual-server.f5.com/health: |
      [
        {
          "path":     "foo.foo.com/",
          "send":     "GET /?monitorf5 HTTP/1.1\r\n",
          "interval": 5,
          "timeout":  10
        }
      ]
    ingress.kubernetes.io/ssl-redirect: "true"
    ingress.kubernetes.io/allow-http: "false"
spec:
  tls:
  - secretName: ************************
  rules:
  - host: foo.foo.com
    http:
      paths:
      - backend:
          serviceName: ******
          servicePort: ******
        path: /

````
if I go to _http://foo.foo.com_ I will be redirected to _https://foo.foo.com_ the same with a / at the end
but if I go to _http://foo.foo.com/toto_ I will stay in http.

If I look to the irule it make sense
Where am I missing something? Isn't it suppose to redirect all request to https which ever the path is?

For a workaround I tried to put /* for the path in the yaml, it works, but in this case I got this error :
`2018/11/06 13:21:28 [ERROR] [2018-11-06 13:21:28,642 f5_cccl.resource.resource ERROR] HTTP error(400): CCCL resource(ApiPolicy) /xxxx/ingress_xxxxxxxxxxxxxx_80.
2018/11/06 13:21:28 [ERROR] [2018-11-06 13:21:28,642 f5_cccl.service.manager ERROR] F5CcclResourceRequestError - 400 Unexpected Error: Bad Request for uri: https://xxxxxxxxxxxxxx:443/mgmt/tm
/ltm/policy/
2018/11/06 13:21:28 [INFO] Text: u'{"code":400,"message":"01071748:3: Policy \'/xxxx/ingress_xxxxxxxxxxxxxx\', rule \'ingress_xxxxxxxxxxxxxx_*_ingress_xxxxxxxxxxxxxx\'; invalid rule name, value \'ingress_xxxxxxxxxxx_*_ingress_xxxxxxxxxxxxxxx\'.","errorStack":[],"apiError":3}'
2018/11/06 13:21:28 [ERROR] [2018-11-06 13:21:28,642 f5_cccl.service.manager ERROR] Resource /xxxx/ingress_xxxxxxxxxxxxxx_80 creation error, requeuing task...`

With the name of the policy it's normal, can you help me on this matter?
#### Kubernetes Version
1.9

#### Controller Version
1.6.0

#### BIG-IP Version
Product     BIG-IP
  Version     12.1.3
  Build       0.0.378
  Edition     Final

#### Diagnostic Information
N/A

--- a/pkg/appmanager/routing.go
+++ b/pkg/appmanager/routing.go
@@ -359,7 +359,7 @@ func httpRedirectIRule(port int32) string {
 				set redir 0
 				foreach s [split $paths "|"] {
 					# See if the request path starts with the prefix
-					append prefix "^" $s "($|/)"
+					append prefix "^" $s "($|/*)"
 					if {[HTTP::path] matches_regex $prefix} {
 						set redir 1
 						break

